enhancedTable/panel-manager.ts

```diff
@@ -20,91 +23,54 @@ export class PanelManager {
-            });
-            if (firstRow === undefined && lastRow === undefined) {
-                firstRow = 0;
-                lastRow = 0;
+    // recursively find and set the legend block for the layer
+    setLegendBlock(legendEntriesList: any) {
+        legendEntriesList.forEach(entry => {
+            if (entry.proxyWrapper !== undefined && this.currentTableLayer._layerProxy === entry.proxyWrapper.proxy) {
+                this.legendBlock = entry;
             }
-            rowRange = firstRow.toString() + " - " + lastRow.toString();
-        }
-        else {
-            rowRange = this.maximized ? '1 - 15' : '1 - 5';
-        }
-        if (this.panel.panelControls.find('.scrollRecords')[0]) {
-            this.panel.panelControls.find('.scrollRecords')[0].innerHTML = rowRange;
-        }
-
-        return rowRange;
+            else if (entry.children || entry.entries) {
+                this.setLegendBlock(entry.children || entry.entries);
+            }
+        });
     }
 
     open(tableOptions: any, layer: any) {
         if (this.currentTableLayer === layer) {
             this.close();
         } else {
             this.tableOptions = tableOptions;
+            this.panelStatusManager = new PanelStatusManager(this);
+            this.panelStatusManager.setFilterAndScrollWatch();
 
             let panelManager = this;
 
-            // when filter is changed, get the correct filter status
-            this.tableOptions.onFilterChanged = function (event) {
-                if (panelManager.tableOptions.api) {
-                    panelManager.tableOptions.api.selectAllFiltered();
-                    panelManager.getFilterStatus();
-                    panelManager.tableOptions.api.deselectAllFiltered();
-                }
-            }
 
-            this.tableOptions.onBodyScroll = function (event) {
-                panelManager.getScrollRange();
-            }
+            // set legend block / layer that the panel corresponds to
+            this.currentTableLayer = layer;
+            this.setLegendBlock(this.currentTableLayer._mapInstance.legendBlocks.entries);
+
+            this.panelRowsManager = new PanelRowsManager(this);
+
 
+            // set header / controls for panel
             let controls = this.header;
             controls = [new this.panel.container('<span style="flex: 1;"></span>'), ...controls];
-            controls = [new this.panel.container(`<div style="padding-bottom :30px"><h2><b>Features: ${layer.name}</b></h2><br><p><span class="scrollRecords">${this.getScrollRange()}</span> of <span class="filterRecords">${this.getFilterStatus()}</span></div>`), ...controls];
+            controls = [new this.panel.container(`<div style="padding-bottom :30px"><h2><b>Features: ${this.legendBlock.name}</b></h2><br><p><span class="scrollRecords">${this.panelStatusManager.getScrollRange()}</span> of <span class="filterRecords">${this.panelStatusManager.getFilterStatus()}</span></div>`), ...controls];
```

Comment:
combine this with the above?
```suggestion
             controls = [
                 new this.panel.container(`<div style="padding-bottom :30px"><h2><b>Features: ${this.legendBlock.name}</b></h2><br><p><span class="scrollRecords">${this.panelStatusManager.getScrollRange()}</span> of <span class="filterRecords">${this.panelStatusManager.getFilterStatus()}</span></div>`), 
                 new this.panel.container('<span style="flex: 1;"></span>'),
                 ...controls
             ];
```

Reply:
donethanks, also updated https://github.com/fgpv-vpgf/plugins/pull/25/files#diff-675df086e13599b4a9bd7a2ca0f289efR105 as per your suggestion